### PR TITLE
Update Vitis AI quantization to support ORT 1.16, support TensorData and QuantizationParams

### PR DESCRIPTION
**Pull request by sheng-xiao**
## Describe your changes
Update Vitis AI quantization to support ORT 1.16, support TensorData and QuantizationParams

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

## (Optional) Issue link
https://github.com/microsoft/Olive/issues/629

### CODE REVIEW

File: olive/passes/onnx/vitis_ai/quantize.py [hunk 1]
```diff
@@ -207,7 +207,20 @@ def quantize_static(
         )
 
         calibrator.collect_data(calibration_data_reader)
-        tensors_range = calibrator.compute_range()
+        try:
+            tensors_range = calibrator.compute_range()
+
+        except AttributeError:
+            tensors_range = calibrator.compute_data()
+            tensors_range_value = {}
```
Comment (guotuofeng) [inline olive/passes/onnx/vitis_ai/quantize.py:215]: would it be more clear if we write 
```python
        if not is_ort_version_below_1_16():
            tensors_range = calibrator.compute_range()
        else:
            tensors_range = calibrator.compute_data()
            tensors_range_value = {}
            for key in tensors_range.data.keys():
                tensors_range_value[key] = tensors_range.data[key].range_value
            tensors_range = tensors_range_value
        
            from onnxruntime.quantization.calibrate import TensorsData

            new_calibrate_tensors_range = TensorsData(CalibrationMethod.MinMax, tensors_range)
            tensors_range = new_calibrate_tensors_range

```

Comment (sheng-xiao) [reply olive/passes/onnx/vitis_ai/quantize.py:214]: I agree and updated, thanks.

**Review by guotuofeng — DISMISSED** [review]

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**